### PR TITLE
Fix QSV presets may be empty

### PR DESCRIPTION
**Pull request by nyanmisaka**
**Changes**
- Fix QSV presets may be empty

**Issues**
- #12630

### CODE REVIEW

**Comment by gnattu** [comment]
I don't think it is necessary after the web part is merged though.

cc @Shadowghost What do you think. Should we do explicit mapping or?

**Comment by Shadowghost** [comment]
I don't think we should do explicit mappings if not necessary. Whole reason to use an enum is to not do this kind of thjngs

**Comment by nyanmisaka** [comment]
I just don't get why the preset can be empty...

**Comment by EraYaN** [comment]
> I don't think we should do explicit mappings if not necessary. Whole reason to use an enum is to not do this kind of things

I feel like explicit mapping is always a better idea, analyzers can alert you to missing cases and you never pass anything that is not explicitly supported.

**Comment by gnattu** [comment]
> I just don't get why the preset can be empty...

It is due to the mistake made in input checking. It checks validity for encoding preset passed in as the function parameter which will always fallback to a valid value `superfast` which always passes the validity check, and then it passes the user config value which could be empty due to the client sent incorrect value which is not a valid enum value, and then passed to ffmpeg as empty string.

File: MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs [hunk 1]
```diff
@@ -1489,16 +1489,16 @@ private string GetEncoderParam(EncoderPreset? preset, EncoderPreset defaultPrese
                      || string.Equals(videoEncoder, "hevc_qsv", StringComparison.OrdinalIgnoreCase) // hevc (hevc_qsv)
                      || string.Equals(videoEncoder, "av1_qsv", StringComparison.OrdinalIgnoreCase)) // av1 (av1_qsv)
             {
-                EncoderPreset[] valid_presets = [EncoderPreset.veryslow, EncoderPreset.slower, EncoderPreset.slow, EncoderPreset.medium, EncoderPreset.fast, EncoderPreset.faster, EncoderPreset.veryfast];
-
-                if (valid_presets.Contains(encoderPreset))
-                {
-                    param += " -preset " + encodingOptions.EncoderPreset;
```
Comment (gnattu) [inline MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs:1496]: Just change this line to pass the encoderPreset as is should be fine if I understand it correctly

Comment (Shadowghost) [reply MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs:1496]: This should likely fix it, yes.

**Review by gnattu — APPROVED** [review]

**Review by Shadowghost — APPROVED** [review]

**Review by crobibero — APPROVED** [review]

**Pull request merged**